04-Authorization/src/Auth/Auth.js

```diff
@@ -42,36 +47,46 @@ export default class Auth {
     });
   }
 
+  getAccessToken() {
+    return this.accessToken;
+  }
+
+  getIdToken() {
+    return this.idToken;
+  }
+
   setSession(authResult) {
+    // Set isLoggedIn flag in localStorage
+    localStorage.setItem('isLoggedIn', 'true');
+
     // Set the time that the access token will expire at
-    let expiresAt = JSON.stringify(
-      authResult.expiresIn * 1000 + new Date().getTime()
-    );
-    // If there is a value on the `scope` param from the authResult,
-    // use it to set scopes in the session for the user. Otherwise
-    // use the scopes as requested. If no scopes were requested,
-    // set it to nothing
-    const scopes = authResult.scope || this.requestedScopes || '';
-
-    localStorage.setItem('access_token', authResult.accessToken);
-    localStorage.setItem('id_token', authResult.idToken);
-    localStorage.setItem('expires_at', expiresAt);
-    localStorage.setItem('scopes', JSON.stringify(scopes));
+    let expiresAt = (authResult.expiresIn * 1000) + new Date().getTime();
+    this.accessToken = authResult.accessToken;
+    this.idToken = authResult.idToken;
+    this.expiresAt = expiresAt;
+
+    // Set the users scopes
+    this.scopes = authResult.scope || this.requestedScopes || '';
+
     // navigate to the home route
     history.replace('/home');
   }
 
-  getAccessToken() {
-    const accessToken = localStorage.getItem('access_token');
-    if (!accessToken) {
-      throw new Error('No access token found');
-    }
-    return accessToken;
+  renewSession() {
+    this.auth0.checkSession({}, (err, authResult) => {
+      console.log(err, authResult);
```

Comment:
Not sure if you agree but we probably shouldn't leave these in (I agree with the ones that print out errors, though).

Reply:
Good suggestion